MatrixSDK/IdentityServer/MXIdentityService.m

```diff
@@ -0,0 +1,511 @@
+    } failure:failure];
+}
+
+- (MXHTTPOperation*)requestPhoneNumberValidation:(NSString*)phoneNumber
+                                     countryCode:(NSString*)countryCode
+                                    clientSecret:(NSString*)clientSecret
+                                     sendAttempt:(NSUInteger)sendAttempt
+                                        nextLink:(NSString *)nextLink
+                                         success:(void (^)(NSString *sid, NSString *msisdn))success
+                                         failure:(void (^)(NSError *error))failure
+{
+    return [self checkAPIVersionAvailabilityAndPerformOperationOnSuccess:^MXHTTPOperation* {
+        return [self.restClient requestPhoneNumberValidation:phoneNumber countryCode:countryCode clientSecret:clientSecret sendAttempt:sendAttempt nextLink:nextLink success:success failure:failure];
+    } failure:failure];
+}
+
+- (MXHTTPOperation *)submit3PIDValidationToken:(NSString *)token
+                                        medium:(NSString *)medium
+                                  clientSecret:(NSString *)clientSecret
+                                           sid:(NSString *)sid
+                                       success:(void (^)(void))success
+                                       failure:(void (^)(NSError *))failure
+{
+    return [self checkAPIVersionAvailabilityAndPerformOperationOnSuccess:^MXHTTPOperation* {
+        return [self.restClient submit3PIDValidationToken:token medium:medium clientSecret:clientSecret sid:sid success:success failure:failure];
+    } failure:failure];
+}
+
+#pragma mark Other
+
+- (MXHTTPOperation *)pingIdentityServer:(void (^)(void))success failure:(void (^)(NSError *))failure
+{
+    return [self checkAPIVersionAvailabilityAndPerformOperationOnSuccess:^MXHTTPOperation* {
+        return [self.restClient pingIdentityServer:success failure:failure];
+    } failure:failure];
+}
+
+- (MXHTTPOperation*)signUrl:(NSString*)signUrl
+                    success:(void (^)(NSDictionary *thirdPartySigned))success
+                    failure:(void (^)(NSError *error))failure
+{    
+    return [self.restClient signUrl:signUrl success:success failure:failure];
+}
+
+#pragma mark - Private
+
+- (MXHTTPOperation*)checkAPIVersionAvailabilityWithSuccess:(void (^)(void))success failure:(void (^)(NSError *))failure
+{
+    if (self.restClient.preferredAPIPathPrefix)
+    {
+        success();
+        return nil;
+    }
+    
+    return [self.restClient isAPIPathPrefixAvailable:kMXIdentityAPIPrefixPathV2 success:^{
+        self.restClient.preferredAPIPathPrefix = kMXIdentityAPIPrefixPathV2;
+        
+        if (success)
+        {
+            success();
+        }
+        
+    } failure:^(NSError * _Nonnull error) {
+        self.restClient.preferredAPIPathPrefix = kMXIdentityAPIPrefixPathV1;
+        
+        if (failure)
```

Comment:
Should we call `success` instead?
If `failure` is called, that will break the requests flow. It seems we want to continue the flow but with V1 api.